src/Optimizer/Configuration/MinifyHtmlConfiguration.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace AmpProject\Optimizer\Configuration;
+
+use AmpProject\Optimizer\Exception\InvalidConfigurationValue;
+use InvalidArgumentException;
+
+/**
+ * Configuration for the MinifyHtml transformer.
+ *
+ * @property bool   $minify                Whether minification is enabled.
+ * @property bool   $minifyAmpScript       Whether amp-script minification is enabled.
+ * @property bool   $minifyJSON            Whether JSON data minification is enabled.
+ * @property bool   $collapseWhitespace    Whether collapsing whitespace is enabled.
+ * @property bool   $removeComments        Whether comments should be removed.
+ * @property bool   $canCollapseWhitespace Whether whitespace can be collapsed.
+ * @property bool   $inBody                Whether the node is in the body.
+ * @property string $commentIgnorePattern  Ignore comments of the form <!-- __AAAA_BBBB___ -->.
```

Comment:
I don't think we should reference anything from Next.js in the PHP toolbox, as all Next.js sites will rather use the NodeJS implementations.

I suggest we do the following here:
- set the pattern to an empty string by default;
- check for the empty string and skip ignoring if found;
- adapt comments to not reference Next.js;
- provide the Next.js pattern in the configuration for the `SpecTest.php` file to ensure we still succeed with the comment ignoring spec test.
- 
```suggestion
 * @property string $commentIgnorePattern  Regex pattern of comments to keep.
```